src/firstboot/firstboot.c

```diff
@@ -1564,7 +1567,27 @@ static int parse_argv(int argc, char *argv[]) {
         return 1;
 }
 
+static int reload_vconsole(sd_bus **bus) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        int r;
+
+        assert(bus);
+
+        if (!*bus) {
+                r = bus_connect_transport_systemd(BUS_TRANSPORT_LOCAL, NULL, RUNTIME_SCOPE_SYSTEM, bus);
+                if (r < 0)
+                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+        }
+
+        r = bus_call_method(*bus, bus_systemd_mgr, "RestartUnit", &error, NULL, "ss", "systemd-vconsole-setup.service", "replace");
+        if (r < 0)
+                return log_error_errno(r, "Failed to issue method call: %s", bus_error_message(&error, r));
```

Comment:
this will only enqueue the job. I wonder if we should wait for it to finish, dunno.

Reply:
Hmm, that's a good point. I copied the code from localed without thinking too much, but here indeed it would be nice to to wait. Although it's a bit risky to wait indefinitely, we should have some (short) timeout.

Reply:
→ https://github.com/systemd/systemd/pull/27787